dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromPrompt.cs

```diff
@@ -281,32 +283,69 @@ protected override async IAsyncEnumerable<TResult> InvokeStreamingCoreAsync<TRes
         {
             asyncReference = textGeneration.GetStreamingTextContentsWithDefaultParserAsync(result.RenderedPrompt, result.ExecutionSettings, kernel, cancellationToken);
         }
+        else if (result.AIService is IChatClient chatClient)
+        {
+            asyncReference = chatClient.GetStreamingResponseAsync(result.RenderedPrompt, result.ExecutionSettings.ToChatOptions(kernel), cancellationToken);
+        }
         else
         {
             // The service selector didn't find an appropriate service. This should only happen with a poorly implemented selector.
-            throw new NotSupportedException($"The AI service {result.AIService.GetType()} is not supported. Supported services are {typeof(IChatCompletionService)} and {typeof(ITextGenerationService)}");
+            throw new NotSupportedException($"The AI service {result.AIService.GetType()} is not supported. Supported services are {typeof(IChatCompletionService)} and {typeof(ITextGenerationService)} and {typeof(IChatClient)}");
         }
 
-        await foreach (var content in asyncReference.ConfigureAwait(false))
+        await foreach (object content in asyncReference.ConfigureAwait(false))
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            yield return typeof(TResult) switch
+            if (content is StreamingKernelContent kernelContent)
```

Comment:
What's our current thinking on abandoning SK's (experimental) content types and just unifying on the M.E.AI.Abstractions ones? Is that something we'll be able to do in the near future?

Reply:
Yes, this will be another phase of changes that we will need to think how it can be minimal impactful (adding extra API's, obsoleting strategies, temporary inheritance). Investigating still..